tests/transcodetests/transcodetests.cc

```diff
@@ -0,0 +1,198 @@
+                                           ::testing::ValuesIn(allFormats)));
+
+bool read_file( string path, void** data, long *fsize ) {
+    FILE *f = fopen(path.data(),"rb");
+    fseek(f, 0, SEEK_END);
+    *fsize = ftell(f);
+    fseek(f, 0, SEEK_SET);  /* same as rewind(f); */
+
+    *data = malloc(*fsize);
+    fread(*data, 1, *fsize, f);
+    fclose(f);
+    return true;
+}
+
+bool isPo2(uint32_t i) {
+    return (i&(i-1))==0;
+}
+
+void test_texture_set( TextureSet & textureSet, FormatFeature & format ) {
+
+    void * basisData;
+    long basisSize;
+    
+    string path = image_path+textureSet.basisuPath;
+    bool read_success = read_file(path, &basisData, &basisSize);
+
+    ASSERT_TRUE(read_success);
+
+    basis_file basisu;
+
+    basisu.open((uint8_t*)basisData,basisSize);
+    uint32_t bWidth = basisu.getImageWidth(0,0);
+    uint32_t bHeight = basisu.getImageHeight(0,0);
+
+    bool hasAlpha = basisu.getHasAlpha();
+
+    ASSERT_EQ(hasAlpha,textureSet.hasAlpha);
+
+    if( !hasAlpha && !format.supportsNonAlpha ) {
+        return;
+    }
+
+    if(!(isPo2(bWidth) && isPo2(bHeight))
+        && !format.supportsNonPo2 ) {
+        return;
+    }
+
+    uint32_t finalSize = basisu.getImageTranscodedSizeInBytes(0,0,format.format);
+    ktx_uint8_t* basisTranscodedData = (ktx_uint8_t*) malloc(finalSize);
+    basisu.startTranscoding();
+    uint32_t bRes = basisu.transcodeImage((void*)basisTranscodedData,finalSize,0,0,format.format,0,0);
+
+    ASSERT_TRUE(bRes);
+
+    basisu.close();
+
+    void * data;
+    long fsize;
+
+    path = image_path+textureSet.ktxPath;
+    read_success = read_file(path, &data, &fsize);
+
+    ASSERT_TRUE(read_success);
+
+    KTX_error_code result;
+    
+    ktxTexture2* newTex = 0;
+    
+    result = ktxTexture2_CreateFromMemory(
+        (const ktx_uint8_t*) data,
+        (ktx_size_t) fsize,
+        KTX_TEXTURE_CREATE_LOAD_IMAGE_DATA_BIT,
+        (ktxTexture2**)&newTex
+        );
+    
+    ASSERT_EQ(result,KTX_SUCCESS);
+
+    result = ktxTexture2_TranscodeBasis(
+                                        newTex,
+                                        format.format,
+                                        0
+                                        );
+    ASSERT_EQ(result,KTX_SUCCESS) << "Format " << format.format;
+
+    EXPECT_EQ(bWidth,newTex->baseWidth);
+    EXPECT_EQ(bHeight,newTex->baseHeight);
+    EXPECT_EQ(finalSize,newTex->dataSize);
+
+    int cmp = std::memcmp(basisTranscodedData,newTex->pData,finalSize);
+
+    ASSERT_EQ(cmp,0);
+
+    ktxTexture_Destroy(ktxTexture(newTex));
+
+    free(data);
+    free(basisTranscodedData);
+
+    free(basisData);
+}
+
+TEST_P(TextureCombinationsTest, Basic) {
+    TextureSet ts = get<0>(GetParam());
+    FormatFeature format = get<1>(GetParam());
+    cout << "txt: " << ts.ktxPath
+            << " format: " << format.format << "\n";
+    test_texture_set(ts,format);
+}
+}  // namespace
+
+int main(int argc, char **argv) {
```

Comment:
Why do you have your own main? In my tests I link with gtest_main.

Reply:
Instead of either copying the final executable into the testimages directory or the other way around I pass the absolute path of the testimages folder as a command line argument.
That's why I use a custom main, to store the path to the images.